nixos/modules/system/boot/stage-1.nix

```diff
@@ -349,6 +349,9 @@ let
       [ { object = bootStage1;
           symlink = "/init";
         }
+        { object = "${modulesClosure}/lib";
+          symlink = "/lib";
+        }
```

Comment:
Do we need to link `/lib/modules` and `/lib/firmware` separately? Since there can be other things in `/lib`.

Reply:
Like what?

Reply:
```nix
        { object = "${modulesClosure}/lib/firmware";
          symlink = "/lib/firmware";
        }
```

Reply:
No, I mean, this path only exists in stage 1, and I don't see anything else in the stage 1 script that touches lib, so what other things that can be in lib are you talking about?

Reply:
> No, I mean, this path only exists in stage 1, and I don't see anything else in the stage 1 script that touches lib, so what other things that can be in lib are you talking about?

I'm not aware of anything currently, but we can always avoid a later footgun.

Reply:
It would be a hard failure though — it's not going to quietly break some future thing.  If we find ourselves needing it (v. unlikely IMO especially since we're switching to systemd stage 1 at some point), it can be changed then.